commands.go

```diff
@@ -3,475 +3,289 @@ package main
-		if m < 1 || m > 9 {
-			b(devbot, "Moves are numbers between 1 and 9!")
-			return
-		}
-		err = tttGame.Apply(tictactoe.Move(m-1), currentPlayer)
-		if err != nil {
-			b(devbot, err.Error())
-			return
-		}
-		b(devbot, "```\n"+tttPrint(tttGame.Cells)+"\n```")
-		if currentPlayer == tictactoe.X {
-			currentPlayer = tictactoe.O
+func roomCommand(u *user, args []string) {
+	if len(args) == 0 {
+		if u.messaging != nil {
+			u.system(fmt.Sprintf("You are currently private messaging %s, and in room %s", u.messaging.name, u.room.name))
 		} else {
-			currentPlayer = tictactoe.X
+			u.system(fmt.Sprintf("You are currently in %s", u.room.name))
 		}
-		if !(tttGame.Condition() == tictactoe.NotEnd) {
-			b(devbot, tttGame.Condition().String())
-			currentPlayer = tictactoe.X
-			tttGame = new(tictactoe.Board)
-			// hasStartedGame = false
-		}
-		return
-	}
 
-	if line == "./users" {
-		b("", printUsersInRoom(u.room))
-		return
-	}
-	if line == "./all" {
-		names := make([]string, 0, len(allUsers))
-		for _, name := range allUsers {
-			names = append(names, name)
+		// Send a list of rooms and the rooms users.
+		type kv struct {
+			roomName   string
+			numOfUsers int
 		}
-		sort.Slice(names, func(i, j int) bool {
-			return strings.ToLower(stripansi.Strip(names[i])) < strings.ToLower(stripansi.Strip(names[j]))
+		var ss []kv
+		for k, v := range rooms {
+			ss = append(ss, kv{k, len(v.users)})
+		}
+		sort.Slice(ss, func(i, j int) bool {
+			return ss[i].numOfUsers > ss[j].numOfUsers
 		})
-		b("", fmt.Sprint(names))
-		return
-	}
-	if line == "easter" {
-		go func() {
-			time.Sleep(time.Second)
-			b(devbot, "eggs?")
-		}()
-		return
-	}
-	if line == "./exit" && !isSlack {
-		u.close(u.name + red.Paint(" has left the chat"))
-		return
-	}
-	if line == "./bell" && !isSlack {
-		u.bell = !u.bell
-		if u.bell {
-			b("", fmt.Sprint("bell on"))
-		} else {
-			b("", fmt.Sprint("bell off"))
+
+		u.system("Rooms and users")
+		for _, kv := range ss {
+			u.system(blue.Paint(kv.roomName) + ": " + printUsersInRoom(rooms[kv.roomName]))
 		}
+
 		return
 	}
-	if strings.HasPrefix(line, "./room") && !isSlack {
-		if u.messaging != nil {
-			u.messaging = nil
-			u.writeln(devbot, "Left private chat")
-			return
-		}
-		rest := strings.TrimSpace(strings.TrimPrefix(line, "./room"))
-		if rest == "" || rest == "s" { // s so "./rooms" works too
-			type kv struct {
-				roomName   string
-				numOfUsers int
-			}
-			var ss []kv
-			for k, v := range rooms {
-				ss = append(ss, kv{k, len(v.users)})
-			}
-			sort.Slice(ss, func(i, j int) bool {
-				return ss[i].numOfUsers > ss[j].numOfUsers
-			})
-			roomsInfo := ""
-			for _, kv := range ss {
-				roomsInfo += blue.Paint(kv.roomName) + ": " + printUsersInRoom(rooms[kv.roomName]) + "  \n"
-			}
-			b("", "Rooms and users  \n"+strings.TrimSpace(roomsInfo))
-			return
-		}
-		if strings.HasPrefix(rest, "#") {
-			if v, ok := rooms[rest]; ok {
-				u.changeRoom(v, sendToSlack)
+	if args[0] == "leave" {
```

Comment:
why do we have this new command, can't `./room` make them leave private chat rooms

Reply:
You might wanna know where people are and see who you are messaging without leaving where you are?

Reply:
@TAG-Epic

Reply:
Great point